app/services/api/declarations/create.rb

```diff
@@ -0,0 +1,142 @@
+module API::Declarations
+  class Create
+    include API::Concerns::Teachers::SharedAction
+
+    attribute :declaration_date
+    attribute :declaration_type
+    attribute :evidence_type
+
+    validates :declaration_date, presence: { message: "Enter a '#/declaration_date'." }
+    validates :declaration_date,
+              future_date: true,
+              declaration_date_within_milestone: true,
+              declaration_date_format: true,
+              allow_blank: true
+    validates :declaration_type, presence: { message: "Enter a '#/declaration_type'." }
+    validates :evidence_type, evidence_type: true
+    validate :validate_only_started_or_completed_if_mentor
+    validate :teacher_not_withdrawn
+    validate :validate_milestone_exists
+    validate :validates_billable_slot_available
+    validate :payment_statement_available
+
+    def create
+      return false unless valid?
+
+      Declarations::Create.new(
+        author: Events::LeadProviderAPIAuthor.new(lead_provider:),
+        lead_provider:,
+        teacher:,
+        training_period:,
+        declaration_date:,
+        declaration_type:,
+        evidence_type:
```

Comment:
Can we clear the evidence type if its not applicable for the declaration; in ECF we do this in the validator:

https://github.com/DFE-Digital/early-careers-framework/blob/main/app/validators/evidence_held_validator.rb#L57

But I think we should do it in this service, Mook added a method for checking if its applicable:

https://github.com/DFE-Digital/register-early-career-teachers-public/blob/main/app/validators/evidence_held_validator.rb#L24-L26